org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -1343,13 +1343,44 @@ protected void load (int ID, String trxName)
 		}
 	}	//	load
 
+	/**
+	 * Load record with UUID
+	 * 
+	 * @param uuID    UUID
+	 * @param trxName transaction name
+	 */
+	public void loadByUU(String uuID, String trxName)
+	{
+		// reset new values
+		m_newValues = new Object[get_ColumnCount()];
+		checkImmutable();
+
+		if (log.isLoggable(Level.FINEST))
+			log.finest("uuID=" + uuID);
+		if (!Util.isEmpty(uuID, true))
+		{
+			load(uuID,trxName);
+		}else {
+			throw new IllegalArgumentException("Invalid null UU - Must pass valid UU");
```

Comment:
the null check should be at the top. since we are throwing illegalargumentexception here, the method shouldn't do anything if uuid is null/empty.

Reply:
Done